test/www/jxcore/test/thalicryptomanagertests.js

```diff
@@ -0,0 +1,61 @@
+require('./mockmobilegetdocpath');
+var test = require('tape');
+var cryptomanager = require('../thali/thalicryptomanager');
+
+
+test('successfully create a new pkcs12 file and return hash value', function (t) {
+  var errorMessage = null,
+      fileLocation = './pkcs12foldergood';
+
+  var hashLength = 16;
+
+  Mobile.storeGetDocumentsPathReturnArguments(errorMessage, fileLocation);
+  
+  cryptomanager.getPublicKeyHash(function (err, publicKeyHash) {
+    t.equal(err, null);
+    t.equal(publicKeyHash.length, hashLength);
+    t.end();
```

Comment:
How do you know that you are actually getting back the proper hash of a properly generated certificate? At a minimum you should read in the pkcs12 file yourself and run the hash function and compare the results so you can be sure.